src/access/wac.test.ts

```diff
@@ -0,0 +1,383 @@
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { jest, describe, it } from "@jest/globals";
+import { IriString, SolidDataset, WithServerResourceInfo } from "../interfaces";
+import { dataset } from "../rdfjs";
+import { getAgentAccess } from "./wac";
+import { Response } from "cross-fetch";
+import { triplesToTurtle } from "../formats/turtle";
+import { mock_addAclRuleQuads } from "../acl/mock.internal";
+import { acl, foaf } from "../constants";
+
+jest.mock("../fetcher.ts", () => ({
+  fetch: jest.fn().mockImplementation(() =>
+    Promise.resolve(
+      new Response(undefined, {
+        headers: { Location: "https://arbitrary.pod/resource" },
+      })
+    )
+  ),
+}));
+
+function getMockDataset(
+  sourceIri: IriString,
+  aclIri?: IriString
+): SolidDataset & WithServerResourceInfo {
+  return Object.assign(dataset(), {
+    internal_resourceInfo: {
+      sourceIri: sourceIri,
+      isRawData: false,
+      linkedResources: {},
+      aclUrl: aclIri,
+    },
+  });
+}
+
+function mockResponse(
+  body?: BodyInit | null,
+  init?: ResponseInit & { url: string }
+): Response {
+  return new Response(body, init);
+}
+
+describe("getAgentAccess", () => {
+  it("calls the included fetcher by default", async () => {
+    const mockedFetcher = jest.requireMock("../fetcher.ts") as {
+      fetch: jest.Mock<
+        ReturnType<typeof window.fetch>,
+        [RequestInfo, RequestInit?]
+      >;
+    };
+
+    const resource = getMockDataset(
+      "https://some.pod/resource",
+      "https://some.pod/resource.acl"
+    );
+    await getAgentAccess(resource, "https://some.pod/profile#agent");
+
+    expect(mockedFetcher.fetch.mock.calls[0][0]).toEqual(
+      "https://some.pod/resource.acl"
+    );
+  });
+
+  it("returns null if no ACL is accessible", async () => {
+    const mockFetch = jest
+      .fn(window.fetch)
+      // No resource ACL available...
+      .mockResolvedValueOnce(
+        mockResponse("", {
+          status: 404,
+          url: "https://some.pod/resource.acl",
+        })
+        // Link to the fallback ACL...
```

Comment:
I'm guessing this might be due to Prettier, but I think this comment and the one below it got with the code following it, i.e. it should be one line down?

Reply:
280cf4d09004624662e0a440c419db280a687d7b